test/test_util.py

```diff
@@ -342,3 +342,74 @@ def test_fix_types(self):
+        # each value of float1, so the 4th value of float1 of -0.4 is reached after
+        # 3 * 18 = 54 values in the generated_grid (and remains the same for the next 18 values):
+        self.assertAlmostEqual(generated_grid[55].get_dictionary()['float1'], -0.4, places=2)
+        # 5th diff. value for int1 after 4 * 3 = 12 values. Reasoning as above.
+        self.assertEqual(generated_grid[12].get_dictionary()['int1'], 63)
+        self.assertEqual(generated_grid[3].get_dictionary()['ord1'], '1')
+        self.assertEqual(generated_grid[4].get_dictionary()['ord1'], '2')
+        self.assertEqual(generated_grid[5].get_dictionary()['ord1'], '3')
+
+        # Tests for quantization and conditional spaces. num_steps_dict supports specifying steps
+        # for only some of the int and float HPs. The rest are taken from the 'q' member variables
+        # of these HPs. The conditional space tested has 2 levels of conditions.
+        cs2 = ConfigurationSpace(seed=123)
+        float1 = UniformFloatHyperparameter(name='float1', lower=-1, upper=1, log=False)
+        int1 = UniformIntegerHyperparameter(name='int1', lower=0, upper=1000, log=False, q=500)
+        cs2.add_hyperparameters([float1, int1])
+
+        int2_cond = UniformIntegerHyperparameter(name='int2_cond', lower=10, upper=100, log=True)
+        cs2.add_hyperparameters([int2_cond])
+        cond_1 = AndConjunction(LessThanCondition(int2_cond, float1, -0.5),
+                                GreaterThanCondition(int2_cond, int1, 600))
+        cs2.add_conditions([cond_1])
+        cat1_cond = CategoricalHyperparameter(name='cat1_cond', choices=['apple', 'orange'])
+        cs2.add_hyperparameters([cat1_cond])
+        cond_2 = AndConjunction(GreaterThanCondition(cat1_cond, int1, 300),
+                                LessThanCondition(cat1_cond, int1, 700),
+                                GreaterThanCondition(cat1_cond, float1, -0.5),
+                                LessThanCondition(cat1_cond, float1, 0.5)
+                                )
+        cs2.add_conditions([cond_2])
+        float2_cond = UniformFloatHyperparameter(name='float2_cond',
+                                                 lower=10., upper=100., log=True)
+        cs2.add_hyperparameters([float2_cond])
+        cond_3 = GreaterThanCondition(float2_cond, int2_cond, 50)
+        cs2.add_conditions([cond_3])
+        # print(cs2)
+        num_steps_dict1 = {'float1': 4, 'int2_cond': 3, 'float2_cond': 3}
+        generated_grid = generate_grid(cs2, num_steps_dict1)
+        # print(generated_grid)
+        self.assertEqual(len(generated_grid), 18)
+
+        # RR: I manually generated the grid and verified the values were correct.
+        # Here, we test that a few randomly chosen values in the generated grid
+        # correspond to the ones I checked.
+        self.assertEqual(generated_grid[3].get_dictionary()['int1'], 1000)
```

Comment:
As above, could you please check maybe the first and last configuration completely?

Reply:
Done